### ISSUE TITLE
Possible to install conflicting versions of opencv alongside each other

### ISSUE THREAD

**Issue by b0ggyb33** [opencv/opencv#24307]
### System Information

OpenCV version: 4.7.0.72 and 4.8.0.76
Operating System: Ubuntu 22.04 and Windows 10
Python version

### Detailed description

It is possible to install opencv-python==4.8.0.76 and opencv-contrib-python-headless==4.7.0.72 alongside each other without pip complaining. Pip should recognise that they are the same project and not allow both packages to be installed.

This causes issues when a project, for example, installs opencv-contrib-python-headless at a specific version and a third party dependency specificies a different version of opencv-python, resulting in two different versions of opencv being installed.

### Steps to reproduce

```bash
pip install opencv-python==4.7.0.72
pip install opencv-contrib-python-headless==4.8.0.76
pip freeze
```

### Issue submission checklist

- [X] I report the issue, it's not a question
- [X] I checked the problem with documentation, FAQ, open issues, forum.opencv.org, Stack Overflow, etc and have not found any solution
- [X] I updated to the latest OpenCV version and the issue is still there
- [ ] There is reproducer code and related data files (videos, images, onnx, etc)

**Comment by asmorkalov** [comment]
OpenCV and OpenCVContrib is the same binary and package in terms of Python. So if you want to use in the same app, then answer is no.

**Comment by b0ggyb33** [comment]
> OpenCV and OpenCVContrib is the same binary and package in terms of Python. So if you want to use in the same app, then answer is no.

Yes, I know, but a third party can have opencv as a requirement, and if I have contrib installed, then pip does not understand that these packages conflict. This is my issue.

**Comment by Adesoji1** [comment]
Thank you, @asmorkalov , for the clarification regarding OpenCV and OpenCVContrib being the same binary/package in Python terms.

To expand on my concern: I believe there's a gap in the way pip handles these packages. If a project has a dependency on a specific version of opencv-contrib-python-headless and another dependency or third-party package requires a different version of opencv-python, the current behavior allows both versions to coexist without warning. This can lead to unforeseen conflicts and unpredictable behavior in the application.

Ideally, pip should recognize this potential conflict and at least issue a warning, if not prevent the installation altogether. This would ensure developers are aware of the conflicting versions and can take appropriate action.

I understand this might be a broader issue with pip itself, or perhaps there's a need to better define these packages' relationships in their respective setup.py or other configuration files.

Would it be appropriate to discuss potential solutions or workarounds here? Or should this be taken up in a different forum or platform, possibly directly with pip maintainers or another related channel?
